runtime/common/ExecutionContext.h

```diff
@@ -61,6 +61,18 @@ class ExecutionContext {
   /// @brief Pointer to simulation-specific simulation data.
   std::unique_ptr<SimulationState> simulationState;
 
+  /// @brief A map of basis-state amplitudes
+  // The list of basis state is set before kernel launch and the map is filled
+  // by the executor platform.
+  std::map<std::vector<int>, std::complex<double>> amplitudeMaps;
```

Comment:
Just curious ... why is this one non-optional but the other new ones in lines 70 and 74 are optional? If amplitudeMaps is optional as well, I'd recommend making that explicit.